code/processes/wdb.q

```diff
@@ -594,6 +592,16 @@ getsortparams:{[]
     ];
     };
 
+idbReload:{[pt]
+    .lg.o[`idb;"starting idb reload"];
+    if[writedownmode in `partbyenum`default;
+        .lg.o[`eod;"initialising wdbhdb for partition: ",string[currentpartition]];
+        $[.proc.proctype~`sort;{ws:exec w from .servers.getservers[`proctype;`wdb;()!();1b;0b];[first ws](`.wdb.initmissingtables;[1b])}[];initmissingtables[0b]];
```

Comment:
what about sortworker processes? do they need to be handled?

why is the arg `1b` if running it from sort process but `0b` if running directly on wdb?

Reply:
I don't believe the sortworker procs will ever be involved here but I could be wrong


When the initmissingtables function was called from the wdb with the new set up, it was called before currentpartition is updated to the next day

The new argument for the function allows the function to be called for either the current value of currentpartition or the next one.

Honestly not too sure on this, so I am open to suggestions

Reply:
> When the initmissingtables function was called from the wdb with the new set up, it was called before currentpartition is updated to the next day

Sorry I missed this comment earlier, I'm not really getting it. Why is it called before currentpartition is updated if called from wdb but after if called from sort? is there any chance of a timing issue here where if sort is very fast it could call it before it gets updated etc.?

this feels quite messy to me

Reply:
It seems to me that if the sorting is done by the sort proc, then the wdb has already moved on and updated currentpartiton, however when the wdb does the sort currentpartition hasn't been updated as the endofday method is blocked

Reply:
so should the initilisation of next partition not be tied to wdb moving to next partition, rather than being tied to idb reload?